keep/common/fedora.py

```diff
@@ -331,6 +337,14 @@ def save(self, logMessage=None):
         'RELS-EXT': 'related objects',
     }
 
+    def update_label(self, object_pid, object_label):
```

Comment:
Thanks for breaking this into a method.  I would recommend renaming it to make it clearer that the method is updating the *pid* label and not the fedora object label.

However, you haven't encapsulated anything.  By making the pid and label parameters that are passed in (even though they are accessible on the object), you've made it useless to extend.  The goal would be that if we have a subclass that needs to pull a title from somewhere besides the MODS to update the pid record, that class would *only* need to extend this method and not touch the main save method that calls it.  Now that I think about it, that actually means that the mods isModified logic and the pid comparison probably also needs to move inside this method.  

Moving the logic into a single object should also make it easier to isolate the functionality for unit testing.